Add instructions for building docs (#567)
vgvassilev/clad: docs/userDocs/source/user/DevelopersDocumentation.rst

  cmake ../clad -DClang_DIR=/usr/lib/llvm-11 -DLLVM_DIR=/usr/lib/llvm-11 -DCMAKE_INSTALL_PREFIX=../inst -DCMAKE_BUILD_TYPE=Debug -DLLVM_EXTERNAL_LIT="``which lit``"
  make && make install

Instructions to build documentation
-------------------------------------

> Make sure you have installed `sphinx` before building clad with documentation.

Reviewer: This does not answer which sphinx version the user needs. What about other Python dependencies? Can we add
a requirements.txt file in the repository for managing python dependencies?

Author: 1) Worked with the latest sphinx version for me should I add the minimum sphinx version required to build the docs ? will need to figure how to find that out

2) I tried adding a requirements.txt file by using the `pipreqs  /path/to/project` command but the file was empty  will try to create the file manually

Reviewer: > The latest sphinx version worked for me

The latest version will change over time :).

> should I add the minimum sphinx version required to build the docs ? will need to figure how to find that out

requirements.txt in the repository should be enough. You can mention in the docs that all the dependencies can be installed as `pip install -r requirements.txt`

> I tried adding a requirements.txt file by using the pipreqs  /path/to/project command but the file was empty will try to create the file manually

Do not create this file manually. The typical way of creating this is as follows:
- Create a virtual environment and source it.
- Install required dependencies.
- Create requirements file using `pip freeze > requirements.txt`

Sphinx Installation `Instructions`_.

.. _Instructions : https://www.sphinx-doc.org/en/master/usage/installation.html

> The following cmake command should be used to build clad along with its 

Reviewer: ```suggestion
The following CMake command should be used to build clad along with its 
```

  cmake ../clad -DClang_DIR=/usr/lib/llvm-11 -DLLVM_DIR=/usr/lib/llvm-11 -DCMAKE_INSTALL_PREFIX=../inst -DCMAKE_BUILD_TYPE=Debug -DLLVM_EXTERNAL_LIT="``which lit``"
  make && make install

Instructions to build documentation
-------------------------------------

Make sure you have installed `sphinx` before building clad with documentation.

> The following command should be used to get all the dependencies to build documentation using `sphinx` :

Reviewer: ```suggestion
Please use the project's `requirements.txt` file to install Python dependencies. This will also install the correct
version of sphinx.
```

.. code-block:: bash

  pip install -r requirements.txt

> Make sure you have installed `doxygen` to build the internal documentation : 

Reviewer: ```suggestion
Make sure `doxygen` is installed if you want to build the internal documentation: 
```

Doxygen Installation `Instructions`_.

.. _Instructions : https://www.doxygen.nl/download.html

> The following CMake command should be used to build clad along with its 

Reviewer: ```suggestion
To build clad and its documentation, use the following CMake command:
```

documentation.

.. code-block:: bash

  cmake ../clad -DClang_DIR=/usr/lib/llvm-11 -DLLVM_DIR=/usr/lib/llvm-11 -DCLAD_INCLUDE_DOCS=ON -DCLAD_ENABLE_DOXYGEN=ON -DCLAD_ENABLE_SPHINX=ON -DCMAKE_INSTALL_PREFIX=../inst -DCMAKE_BUILD_TYPE=Debug -DLLVM_EXTERNAL_LIT="``which lit``"

>   make sphinx-clad

Reviewer: Can you add where users can find the build documentation?

Also, can they build just doxygen documentation or just sphinx documentation?
Does the `sphinx-clad` target by default builds both the documentation?
